src/Domain/SubscriptionRepository.php

```diff
@@ -11,6 +11,14 @@
  */
 interface SubscriptionRepository {
 
+	/**
+	 * Add or update a subscription.
+	 *
+	 * If the subscription exists in the repository, update it.
+	 *
+	 * @param Subscription $subscription
+	 * @return mixed
```

Comment:
#153

We should also get to documenting exceptions for the repos. For these it's real important that we don't end up catching Doctrine ones in the domain or usecases.